File: js/examples/zcsv_print.ts
import { ZippedCsv } from "../src/index";

const args = process.argv.slice(2);
if (args.length < 2) {
  console.error("Usage: tsx examples/zcsv_print.ts <path_to_zipped_csv_file> <table_index>");
  process.exit(1);
}

const tableIndex = parseInt(args[1], 10);
if (isNaN(tableIndex) || tableIndex < 0) {
  console.error("Table index must be a non-negative integer.");
  process.exit(1);
}

try {
  const zcsv = new ZippedCsv(args[0]);
  const tables = zcsv.getTableNames();

  if (tableIndex >= tables.length) {
    console.error(`Table index ${tableIndex} out of range. Archive has ${tables.length} table(s).`);
    process.exit(1);
  }

  const name = tables[tableIndex];
  const csv = zcsv.getCsv(name);

  console.log(`Table: ${name}`);
  console.log(`Header: ${JSON.stringify(csv.getHeader())}`);
  for (let i = 0; i < csv.totalRows(); i++) {
    console.log(`Row ${i}: ${JSON.stringify(csv.getRow(i))}`);
  }
} catch (err) {
  console.error(`Error: ${err}`);
  process.exit(1);
}

File: cpp/examples/zcsv_print.cpp
// zcsv_print — prints the contents of all (or a named) table in a .zcsv archive.
//
// Usage: zcsv_print <archive.zcsv> [<table_name>]

#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>

#include "csv.h"
#include "zippedcsv.h"

static void PrintTable(const std::string& name, mc::Csv& csv) {
  std::cout << "=== " << name << " ===\n";
  auto header = csv.GetHeader();
  for (size_t i = 0; i < header.size(); ++i) {
    if (i > 0) std::cout << "\t";
    std::cout << header[i];
  }
  std::cout << "\n";

  for (int r = 0; r < csv.TotalRows(); ++r) {
    auto row = csv.GetRow(r);
    for (size_t i = 0; i < header.size(); ++i) {
      if (i > 0) std::cout << "\t";
      std::cout << row[header[i]];
    }
    std::cout << "\n";
  }
  std::cout << "\n";
}

int main(int argc, char* argv[]) {
  if (argc < 2) {
    std::cerr << "Usage: zcsv_print <archive.zcsv> [<table_name>]\n";
    return EXIT_FAILURE;
  }

  std::string path(argv[1]);
  std::string filter = argc >= 3 ? std::string(argv[2]) : "";

  mc::ZippedCsv z(path);
  auto names = z.GetTableNames();

  if (names.empty()) {
    std::cout << "Archive contains no tables.\n";
    return EXIT_SUCCESS;
  }

  for (const auto& name : names) {
    if (!filter.empty() && name != filter && name != filter + ".csv") continue;
    try {
      PrintTable(name, z.GetCsv(name));
    } catch (const std::exception& e) {
      std::cerr << "Error reading '" << name << "': " << e.what() << "\n";
    }
  }

  return EXIT_SUCCESS;
}
